### PR TITLE
[Split Checkout] Stop horizontal scroll on body in order summary page

### LINKED ISSUE #9976: [Split Checkout] Cart and order summary don't fit to screen on mobile devices
## Description

The layout of the order summary table on pages cart and order summary is too wide to fit the screen of mobile devices. Especially the very long names of applied enterprise fees could be a problem here.

## Expected Behavior

The layout should fit the screen without scrolling horizontally.

## Actual Behaviour

The layout doesn't fit the screen and the user needs to scroll horizontally or zoom out.

## Steps to Reproduce

1. Use a mobile device or Browserstack
2. Make sure split checkout is activated
3. Proceed to checkout or to cart
4. Observe the cart and order summary table

## Animated Gif/Screenshot

Chrome on Galaxy S22
image

## Workaround

Zoom or scroll horizontally.

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used: https://github.com/openfoodfoundation/openfoodnetwork/commit/70e47f3929eaed07b951adca36b12b2fbbfc8728
* Browser name and version: Firefox mobile, Safari mobile, Chrome mobile
* Operating System and version (desktop or mobile): Android, iOS

### PR DESCRIPTION
**Pull request by binarygit**
#### What? Why?

- Closes #9976 

#### What should we test?

- Login as user and populate cart with some products (with long names)
- Navigate to checkout. Make sure you've split checkout enabled.
- Ascertain in the `Order Summary` page, the body doesn't scroll

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by binarygit** [comment]
I made the table that contains the order summary scroll-able, instead of the body overflowing and the user having to scroll the body to see the complete summary.

If there's another solution for this that you think is appropriate, let me know please :smile:

**Review by jibees — APPROVED** [review]
Simple, should do the job ;) 👍

**Comment by RachL** [comment]
hello @binarygit the original issue cover both the order summary page and the cart page. What do you prefer: apply the changes here in the cart page as well or create a second PR?

**Comment by binarygit** [comment]
Hey @RachL I fixed the scroll on the cart page in this PR itself.

**Comment by abdellani** [comment]
@binarygit

by "card page" do you refer to "payment method"?  I checked that page and I didn't see a problem.

**Comment by RachL** [comment]
@abdellani it was the `/cart` page

**Comment by RachL** [comment]
@binarygit thanks for the edit, the scroll is working fine! But there is an overlap now with the footer and the last button (I've checked Chrome, Safari and Firefox - happens on all 3):

image

can you fix this bit?

**Comment by binarygit** [comment]
will fix this @RachL. Thanks for pointing it out.

**Comment by binarygit** [comment]
@RachL I fixed it.

File: app/webpacker/css/darkswarm/cart-page.scss [hunk 1]
```diff
@@ -6,6 +11,8 @@
 
 #cart-detail {
   width: 100%;
+  display: block;
```
Comment (jibees) [inline app/webpacker/css/darkswarm/cart-page.scss:14]: This one is necessary? I'm mean, it's not `block` by default?

**Review by jibees — COMMENTED** [review]
I've two comments... Thanks @binarygit !

**Review by abdellani — APPROVED** [review]
I don't have anything to add after the comments from @jibees 
Thank you @binarygit

File: app/views/spree/orders/edit.html.haml [hunk 2]
```diff
@@ -25,7 +25,7 @@
   .sub-header.show-for-medium-down
     = render partial: "shopping_shared/order_cycles"
 
-  %fieldset.footer-pad
+  #cart-container
```
[the remark this answers is not in the document]
Comment (binarygit) [reply app/views/spree/orders/edit.html.haml:28]: Yup I think it was useless here which is why I replaced it with a div.

File: app/webpacker/css/darkswarm/cart-page.scss (hunk 1 above)
Comment (binarygit) [reply app/webpacker/css/darkswarm/cart-page.scss:14]: No, `cart-detail` is a table which is why it's not a `block` by default. And only when I give it a display of `block` I'm able to make the overflow work correctly.

**Review by jibees — APPROVED** [review]

**Comment by RachL** [comment]
@binarygit it's weird now I have again an horizontal scroll on the cart page but only on android with chrome, firefox (iphone is working fine) :eyes:

image

But again we can merge as is and fix in a separated PR?

**Not related to this PR**

On iphone (safari + chrome) however, the whole split checkout seems to have an horizontal scroll (no page fit the screen).

@jibees I will open a dedicated issue for this but I've noticed with this test that split checkout is not at all responsive on iphone :scream: I will retest when #10146 hits test ready. But here are my notes for the record:

image

image

image

See android in comparison: 

image

**Comment by binarygit** [comment]
@RachL hmm strange...not sure what's wrong at all. When using dev tools to simulate mobile view, everything is working fine :thinking: but sure let's merge this and deal with the new issues in a separate PR.

**Pull request merged**